docs/grammar/Solidity.g4

```diff
@@ -368,7 +368,10 @@ numberLiteral: (DecimalNumber | HexNumber) NumberUnit?;
 /**
  * A curly-braced block of statements. Opens its own scope.
  */
-block: LBrace statement* RBrace;
+block:
+	LBrace ( statement | uncheckedBlock )* RBrace;
+
+uncheckedBlock: Unchecked block;
```

Comment:
This allows for nested unchecked blocks.

Reply:
Yes, it is not a parser but a syntax checker restriction.

Reply:
Is ``if (<sth>) unchecked { x; }`` invalid? (I'll probably know when I'm done reviewing...)

Reply:
Yeah, apparently it isn't - not sure if it couldn't be valid, but fine!